### PR TITLE
KDB.AI

### PR DESCRIPTION
**Pull request by nbyrneKX**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        self,
+        table: Any = None,
+        hybrid_search: bool = False,
+        sparse_encoder: Optional[Callable] = None,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        try:
+            import kdbai_client as kdbai
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py:83]: this should be added as a dependency in the pyproject.toml

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/utils.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+def default_sparse_encoder(texts: list[str]) -> list[dict[int, int]]:
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/utils.py:1]: typing should use the `from typing import List, Dict, ..` types, to work with python3.8

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+import pandas as pd
+
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.schema import BaseNode, TextNode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from .utils import default_sparse_encoder
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py:19]: lets use the full import here `llama_index.vector_stores.kbdai.utils`

**Comment by logan-markewich** [comment]
@nbyrneKX just a slight bump on the review comments I left

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py (hunk 1 above)
Comment (nbyrneKX) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py:83]: As far as i can tell this is already included.  Did you mean somewhere else?
* https://github.com/run-llama/llama_index/blob/6f04122bc89e8e96afe2da198238219a72496cc3/llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/pyproject.toml#L39

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py (hunk 3 above)
Comment (nbyrneKX) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py]: Updated

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/utils.py (hunk 2 above)
Comment (nbyrneKX) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/utils.py]: Updated

File: llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py:83]: Oh I must have missed that

**Comment by nbyrneKX** [comment]
> @nbyrneKX just a slight bump on the review comments I left

@logan-markewich I noticed the pipeline failed on formatting issue which has now been resolved.  Hopefully that is the last change required

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**